src/components/query-panel/QueryPanelCtrl.ts

```diff
@@ -0,0 +1,74 @@
+import CodeMirror = require("codemirror");
+import {RestService} from "../../services/rest/RestService";
+import {EditorConfiguration} from "codemirror";
+
+export class QueryPanelCtrl implements ng.IComponentController {
+  static $inject: string[] = ["restService", "$timeout"];
+
+  cacheName: string;
+  name: string;
+  description: string;
+  statusCode: number;
+  editor: CodeMirror.Editor;
+  panelType:string;
+
+  constructor(private restService: RestService, private $timeout: ng.ITimeoutService) {
+  }
+
+  executeAction(): void {
+    let inputParam1: string = angular.element(document.querySelector("#" + this.name + "-input-text")).val();
+    let inputParam2: string = angular.element(document.querySelector("#" + this.name + "-input-text-area")).val();
```

Comment:
Is this not an anti-pattern? We should be passing the input values to the QueryPanel component. There are many example of this in our code.

Reply:
But of course it is. Making corrections. Thanks @ryanemerson